src/Streamly/Coreutils/Id.hs

```diff
@@ -0,0 +1,56 @@
+-- |
+-- Module      : Streamly.Coreutils.Id
+-- Copyright   : (c) 2022 Composewell Technologies
+-- License     : BSD-3-Clause
+-- Maintainer  : streamly@composewell.com
+-- Stability   : experimental
+-- Portability : GHC
+--
+-- Get real and effective user and group IDs.
+
+module Streamly.Coreutils.Id
+    (
+      getRealUserID
+    , getRealGroupID
+    , getEffectiveUserID
+    , getEffectiveGroupID
+    , getGroups
+    , getLoginName
+    , getEffectiveUserName
```

Comment:
Why don't we have something equivalent to the `id` command with options here? Why do we have separate functions?